### PR TITLE
Change signature of `GenerativeFunction.edit` to always pass argdiffs (2/N)

### PR DESCRIPTION
**Pull request by femtomc**
Builds on #1322 -- changes the signature of `GenerativeFunction.edit` to always pass the argdiffs (which, (a) is actually correct and makes sense compositionally with respect to the returned "backwards edit request" in the signature of the interface and (b) in general, is required for more advanced edit request programming, where edit requests might choose to validate their own application by considering argdiffs passed to them).

### CODE REVIEW

File: src/genjax/_src/core/generative/generative_function.py
Comment (sritchie) [inline src/genjax/_src/core/generative/generative_function.py]: what is this error showing?

File: src/genjax/_src/core/generative/generative_function.py
Comment (sritchie) [inline src/genjax/_src/core/generative/generative_function.py]: codecov giving us the goods, let's add tests that hit these if possible. Smaller PRs make this a breeze

**Review by sritchie — APPROVED** [review]
Don't merge until we merge the first one, then we can merge them into main one by one. I'll show you how to do it easily with the squash merges

File: src/genjax/_src/core/generative/generative_function.py
Comment (femtomc) [reply src/genjax/_src/core/generative/generative_function.py]: this error is showing that beartype can't figure out forward references

**Pull request closed without merging**
